replication/README.md

```diff
@@ -390,6 +390,9 @@ message GetVolumeReplicationInfoResponse {
   // This field is OPTIONAL.
   // The value of this field MUST NOT be negative.
   int64 last_sync_bytes = 3;
+  // Holds the latest replication state.
+  // This field is REQUIRED.
+  string replication_state = 4;
```

Comment:
This probably should be optional instead of required. If it is a message, maybe call it `status_message` instead, and explain that it is user-facing information.

If something is expected to inspect the contents and act on it, like a _state_, it definitely should be an enum that has values which fit for different storage providers.

Reply:
> If it is a message, maybe call it status_message instead, and explain that it is user-facing information.

It would be the mirroring status state like `stopped` , `syncing`, `error` `replaying` and so on. Not sure, if we should make it `status_message` instead.

> If something is expected to inspect the contents and act on it, like a state, it definitely should be an enum that has values which fit for different storage providers.

Based on the above mentioned value, I am not sure which values to use that make it fit for different storage providers?
Mkaing it generic like `primary`, `secondary`, `unkown` and `error` could be one way

Reply:
And, then the storage provider is responsible for mapping the respective states for them to the csi-addons spec's state?
Like, for us (ceph's pov) `stopped` to `primary`, `replaying` to `secondary`, `error` to `error` and any other state to `unkown`

Reply:
sounds reasonable, each state should have a description, something along these lines:

0. `unknown` status was not provided
1. `primary` this volume is in a primary state and can be used for writing to
2. `secondary` this volume is actively tracking changes from a primary volume
3. `error` something went wrong, and replication has stopped

The `error` state is probably final, and no automated recovery is possible. Maybe a mis-configuration or environment failure.

In case there is an error state that can be recovered automatically (or retried), such a state should be mentioned separately, maybe something like `split-brain` or `out-of-space`.

Reply:
@nixpanic IMO we should keep it as string because it will be hard to map this from storage to the csi-addons and also state is more of enum where we have pre defined states .  can we do something like below


```go
// Represents the current replication state as reported by the backend storage system.
//
// This field is optional.
//
// This is a free-form string that conveys the current replication state,
// such as "synchronized", "initializing", "error", "paused", etc.
//
// Storage providers SHOULD use values that make sense within their own ecosystem.
// This field is intended to be displayed to users or for diagnostic purposes,
// and is NOT intended to be used for strict programmatic logic across vendors.
string replication_status = 4;
```

Reply:
+1 for status_message as we need this to be representation only for users for information not the programmatic

Reply:
We can have status and a message for the very clear status and the more details message as well
```
enum Status {
    UNKNOWN = 0;
    HEALTHY = 1;
    DEGRADED = 2;
    ERROR = 3;
  }

 Status status = 1;
 string status_message = 2
```

Reply:
Updated the RPC based on the above discussion, please take a look now.